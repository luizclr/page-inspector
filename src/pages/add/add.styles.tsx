import styled from "styled-components";

export const Title = styled.h2`
  font-size: 18pt;
  margin-bottom: 20px;
  margin-left: 30px;
  color: #474b4e;
`;
