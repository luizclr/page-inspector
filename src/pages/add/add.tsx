import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

import { Button, Input, Title, Wrapper } from "~/pages/add/add.styles";
import { useInspection } from "~/providers/inspection/inspection";
import { InspectionActionTypes } from "~/reducers/inspector/types";
import { isEmpty } from "~/utils";

export const Add: React.FC = () => {
  const [text, setText] = useState<string>("");
  const { inspectionService, inspectionDispatch } = useInspection();
  const navigate = useNavigate();

  useEffect(() => {}, []);

  const handleClick = async (): Promise<void> => {
    try {
      inspectionDispatch({
        type: InspectionActionTypes.loading,
        payload: {
          isLoading: true,
        },
      });
      const { data } = await inspectionService.addInspection({ keyword: text });

      setText("");
      navigate("/search", {
        state: {
          key: text,
          id: data.id,
        },
      });
    } catch (error) {
      inspectionDispatch({
        type: InspectionActionTypes.error,
      });
    }
  };

  return (
    <div>
      <Title>Add</Title>
      <Wrapper>
        <Input
          type="text"
          placeholder="e.g: security"
          value={text}
          onChange={(e) => {
            setText(e.target.value);
          }}
        />
        <Button disabled={isEmpty(text)} onClick={handleClick} type="submit">
          ADD
        </Button>
      </Wrapper>
    </div>
  );
};
