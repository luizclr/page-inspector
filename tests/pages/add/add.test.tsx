import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { Add } from "~/pages/add/add";

describe("<AddSearch />", () => {
  it("should be able to mount <AddSearch /> component", async () => {
    // given
    const { asFragment } = render(
      <BrowserRouter>
        <Add />
      </BrowserRouter>
    );

    // when
    const sut = asFragment();

    // then
    expect(sut).toMatchSnapshot();
  });
});
