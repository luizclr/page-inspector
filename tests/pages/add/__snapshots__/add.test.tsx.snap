// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddSearch /> should be able to mount <AddSearch /> component 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="sc-bcXHqe jtSBjQ"
    >
      Add
    </h2>
    <div
      class="sc-gswNZR jVxbqf"
    >
      <input
        class="sc-dkrFOg fvqUtP"
        placeholder="e.g: security"
        type="text"
        value=""
      />
      <button
        class="sc-hLBbgP iamWso"
        disabled=""
        type="submit"
      >
        ADD
      </button>
    </div>
  </div>
</DocumentFragment>
`;
